ssg/build_cpe.py

```diff
@@ -175,15 +174,24 @@ class CPEALPlatformSpecification(object):
     prefix = "cpe-lang"
     ns = PREFIX_TO_NS[prefix]
 
-    def __init__(self):
+    def __init__(self, cpe_products):
```

Comment:
This class basically parses platform definitions into expressions, and maintains a list of distinct platforms. This functionality could be relocated to a lower level - a Rule data structure could contain the parsed expression (i.e. the logical tree) already. Moreover, having a list of platforms close to the list of rules would make sense as well.